### PR TITLE
Project access control

### PR DESCRIPTION
**Pull request by EliasEriksson**
New stuff:
* access Engine
* updated project interface to include activities and access

Fixes:
* rearanged fixtures in the entire project
* moved `weekmeter.Rule` and `weekmeter.Rules` udner the namespace `weekmeter.Time`

### CODE REVIEW

**Review by EliasEriksson — COMMENTED** [review]
self check

File: Code/fixtures.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+	const mod =
+		options?.n == undefined
+			? 1
+			: options.resource == "organization"
+			? 7
+			: options.resource == "client"
+			? 5
+			: options.resource == "project"
+			? 3
+			: 2
```
Comment (LinusDahlstrom) [inline Code/fixtures.ts:13]: i know this is just fixtures but is there a thinking behind the integers 1, 7, 5 , 3 ,2 ?

File: Project/Access/Engine/State.spec.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// import from root when done
```
Comment (LinusDahlstrom) [inline Project/Access/Engine/State.spec.ts]: ?

File: Project/Access/Engine/State.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	user: { email: string; [property: string]: string | undefined }
+}
+export namespace State {
+	export const type = isly.object<State>({
+		project: isly.object<State["project"]>({
+			code: Code.type,
+			name: isly.string(),
+			client: isly.object<State["project"]["client"]>({
+				code: Code.type,
+				name: isly.string(),
```
Comment (LinusDahlstrom) [inline Project/Access/Engine/State.ts]: isly has a .pick() method that I think can be used instead.

File: Code/fixtures.ts (hunk 1 above)
Comment (EliasEriksson) [reply Code/fixtures.ts:13]: it is used to generate predictable ids for the resources using codes. These numbers have just been moved from the previous root fixture file. It allows us to generate a client with a few projects that have a few activities etc easily.

File: Project/Access/Engine/State.spec.ts (hunk 2 above)
Comment (EliasEriksson) [reply Project/Access/Engine/State.spec.ts]: changed the import. this is fixed

File: Project/Access/Engine/State.ts (hunk 3 above)
Comment (EliasEriksson) [reply Project/Access/Engine/State.ts]: :+1:

**Review by LinusDahlstrom — APPROVED** [review]

**Pull request merged**